### PR TITLE
reduce all major allocations in replication path

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
reduce all major allocations in replication path

## Motivation and Context
- remove targetClient for passing around via replicationObjectInfo{}
- remove cloning to object info unnecessarily
- remove objectInfo from replicationObjectInfo{} (only require necessary fields)

## How to test this PR?
CI/CD should cover it

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/bucket-replication-utils.go [hunk 1]
```diff
@@ -301,35 +310,23 @@ func parseReplicateDecision(ctx context.Context, bucket, s string) (r ReplicateD
...
-		sync, err = strconv.ParseBool(tgt[1])
-		if err != nil {
-			return r, err
-		}
-		tgtClnt := globalBucketTargetSys.GetRemoteTargetClient(slc[0])
-		if tgtClnt == nil {
-			// Skip stale targets if any and log them to be missing atleast once.
-			logger.LogOnceIf(ctx, fmt.Errorf("failed to get target for bucket:%s arn:%s", bucket, slc[0]), slc[0])
-			// We save the targetDecision even when its not configured or stale.
+		if globalBucketTargetSys.GetRemoteTargetClient(tgt[2]) != nil {
```
Comment (poornas) [inline cmd/bucket-replication-utils.go]:
```suggestion
		if globalBucketTargetSys.GetRemoteTargetClient(tgt[2]) == nil {
```

File: cmd/bucket-replication-utils.go [hunk 2]
```diff
@@ -549,14 +559,8 @@ func getHealReplicateObjectInfo(objInfo ObjectInfo, rcfg replicationConfig) Repl
...
-func (ri *ReplicateObjectInfo) getReplicationState() ReplicationState {
-	rs := ri.ObjectInfo.getReplicationState()
-	rs.ReplicateDecisionStr = ri.Dsc.String()
-	return rs
-}
-
-// vID here represents the versionID client specified in request - need to distinguish between delete marker and delete marker deletion
-func (o *ObjectInfo) getReplicationState() ReplicationState {
+// ReplicationState - returns replication state using other internal replication metadata in ObjectInfo
+func (o ObjectInfo) ReplicationState() ReplicationState {
```
Comment (poornas) [inline cmd/bucket-replication-utils.go:560]:
```suggestion
func (oi ObjectInfo) ReplicationState() ReplicationState {
```

**Review by poornas — CHANGES_REQUESTED** [review]

**Comment by poornas** [comment]
lint issues @harshavardhana

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by poornas — APPROVED** [review]

**Pull request merged**